src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -520,7 +538,60 @@ private void checkWhetherMetadataExistsBeforeSave(SettingConfiguration entity, L
 				settingsMetadataList.add(metadata);
 			}
 		}
-			settingMetadataMapper.insertMany(settingsMetadataList);
+		insertSettingMetadata(settingsMetadataList);
+	}
+
+	private void insertSettingMetadata(List<SettingsMetadata> settingsMetadataList) throws SQLException {
+		String insertSettingMetadata = "INSERT INTO core.settings_metadata ( settings_id, document_id, identifier, "
+				+ "server_version, team, team_id, provider_id, location_id, uuid, json, setting_type, setting_value, setting_key, setting_description, setting_label, inherited_from) VALUES(?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
```

Comment:
This should handle cases where the settings_metadata exist.  Add https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT clause that will update an existing record if its found. 

Currently I have noticed problems where metadata is not updated if does exist

Reply:
How are you send in the update payload? The update doesn't pass through this method.

Reply:
tested by posting to v1 sync API which is used to post settings from CSV on webUI

Reply:
I think this is a different issue. It's not affected by this.  I also just tested it. I will be fixing it too.